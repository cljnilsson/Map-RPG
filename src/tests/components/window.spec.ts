import { expect, test } from "vitest";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/svelte";
import Window from "$lib/features/window/window.svelte";
import { fireEvent } from "@testing-library/svelte";

const renderOptions = {
	width: 400,
	height: 600,
	x: 10,
	y: 10,
	toggleKey: "x",
	visibility: true,
	uniqueKey: "test-window",
};

test("Window renders", async () => {
	const { component, container } = render(Window, renderOptions);

	expect(component).toBeTruthy();
	expect(container).toMatchSnapshot();
});

test("Window closes by click", async () => {
	const user = userEvent.setup();

	render(Window, renderOptions);

	const buttons = screen.getAllByRole("button");

	expect(buttons[0]).toBeVisible();

	await user.click(buttons[3]); // Close button
	//await fireEvent.click(buttons[3]);

	expect(buttons[0].closest(".overlay-rect")).not.toBeVisible();
});

test("Window closes by keybind", async () => {
	//const user = userEvent.setup();

	render(Window, renderOptions);

	const buttons = screen.getAllByRole("button");

	expect(buttons[0]).toBeVisible();

	// user.keyboard adds 100~ms to the test so using fireEvent instead for now

	//await user.keyboard("y");
	await fireEvent.keyDown(document.body, { key: "y" });

	expect(buttons[0].closest(".overlay-rect")).toBeVisible();

	//await user.keyboard("x");
	await fireEvent.keyDown(document.body, { key: "x" });

	expect(buttons[0].closest(".overlay-rect")).not.toBeVisible();
});
