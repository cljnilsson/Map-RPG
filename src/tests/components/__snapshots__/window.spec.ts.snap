// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Window renders 1`] = `
<div>
  <!---->
  <div
    class="overlay-rect svelte-110qxpm"
    style="left: 4px; top: 5.333333333333333px; width: 400px; z-index: 499;"
  >
    <div
      role="button"
      style="cursor: grab;"
      tabindex="0"
    >
      <div
        class="row wrapper svelte-yibhvk"
      >
        <div
          class="col"
        >
          <div
            class="row align-items-center"
          >
            <div
              class="col"
            >
              <!---->
            </div>
             
            <div
              class="col-auto text-end"
            >
              <button
                aria-label="Lock/Unlock"
                class="btn btn-sm btn-outline-secondary"
              >
                <i
                  class="bi bi-lock-fill svelte-110qxpm"
                />
              </button>
              <!---->
               
              <button
                aria-label="Minimize"
                class="btn btn-sm btn-outline-secondary"
              >
                <i
                  class="bi bi-dash svelte-110qxpm"
                />
              </button>
              <!---->
               
              <button
                aria-label="Close"
                class="btn btn-sm btn-outline-secondary"
              >
                <i
                  class="bi bi-x"
                />
              </button>
              <!---->
            </div>
            <!---->
          </div>
           
          <div
            class="row"
          >
            <div
              class="col-auto"
            >
              <small
                class="me-1"
              >
                x: 4
              </small>
              <small
                class="mx-1"
              >
                y: 5
              </small>
               
              <small
                class="mx-1"
              >
                h: 600
              </small>
               
              <small
                class="mx-1"
              >
                w: 400
              </small>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <!---->
    </div>
    <!---->
     
    <div
      class="content-wrapper svelte-110qxpm"
      style="height: 600px;"
    >
      <div
        class="inner d-flex flex-column h-100 svelte-110qxpm"
      >
        <!---->
         
        <!---->
      </div>
    </div>
  </div>
  
  
</div>
`;
